numpy/f2py/cfuncs.py

```diff
@@ -572,18 +572,20 @@
 """
 cppmacros["F2PY_THREAD_LOCAL_DECL"] = """\
 #ifndef F2PY_THREAD_LOCAL_DECL
-#if defined(_MSC_VER) \\
-      || defined(_WIN32) || defined(_WIN64) \\
```

Comment:
Is there any reason to no longer check for `_WIN32` and `_WIN64`? Is `_MSC_VER` set for other compilers on Windows?

Relevant Reference:
[1] [Predefined macros for Microsoft](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros)

Reply:
This was added only in gh-18910, and it feels wrong. There shouldn't be a distinction between 32-bit and 64-bit Windows, and it is not clear that other compilers actually need `__declspec(thread)` instead of one of the alternatives.